pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -174,7 +175,8 @@ public String consumerName() {
                         .build();
 
                 ByteBuf metadataAndPayload = entry.getDataBuffer();
-
+                // increment ref-count of data and release at the end of process: so, we can get chance to call entry.release
+                entry.retain();
```

Comment:
The semantic here is kind of weird, it's kind of mix-matching ref-counting and recycler